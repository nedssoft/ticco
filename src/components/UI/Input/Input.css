.Input {
  width: 100%;
  box-sizing: border-box;
}

.Label {
  font-weight: bold;
  display: block;
  margin-bottom: 5px;
  box-sizing: border-box;
  text-align: left;
  width: 100%;
  margin-left: 10%;
}

.InputElement{
  outline: none;
  width: 80%;
  border: 1px solid #ccc;
  background: #fff;
  padding: 6px 10px;
  font: inherit;
  display: block;
  margin: 10px auto;
  box-sizing: border-box;
  font-size: 18px;
}

.InputElement:focus {
  outline: none;
  background: #ccc;
}

.Invalid {
  border: 1px solid red;
  background-color: salmon;
}

.Valid {
  border: 1px solid rgb(106, 161, 106)
}
