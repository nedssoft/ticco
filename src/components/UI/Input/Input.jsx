import React from 'react'

import styles from './Input.css'


const Input = (props) => {
  const { 
    elementType, 
    elementConfig, 
    value, 
    label, 
    changed,
    valid,
    shouldValidate,
    touched
  } = props;
  const { id } = elementConfig
  let inputElement = null
  const inputClasses = [styles.InputElement]

  if (!valid && shouldValidate && touched) {
    inputClasses.push(styles.Invalid)
  }
  if (valid) {
    inputClasses.push(styles.Valid)
  }
  switch (elementType) {
    case 'input':
      inputElement = (
        <input 
          {...elementConfig}
          className={inputClasses.join(' ')}
          onChange={changed}
          value={value}
          id={id}
        />
      );
      break;
    case 'textarea':
      inputElement = (
        <textarea 
          {...elementConfig} 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
          id={id}
        />
      )
      break;
      case 'select':
      inputElement = (
        <select 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
          id={elementConfig.id}
          label={elementConfig.label}
        >
          {elementConfig.options.map(option => (
            <option value={option.value} key={option.value}>
              {option.displayValue}
            </option>
          ))} 
        </select>
      )
      break;
    default:
      inputElement = (
        <input 
          {...elementConfig} 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
        />
      )
}
  return (
    <div className={styles.Input}>
      <label htmlFor={id} className={styles.Label}> {label}</label>
      { inputElement }
    </div>
  )
}

export default Input
