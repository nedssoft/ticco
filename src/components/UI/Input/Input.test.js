import React from 'react'
import { shallow } from 'enzyme'
import Input from './Input'

describe('<Input />', () => {
  const props = {
      elementType: 'input',
        elementConfig: {
          placeholder: 'Password',
          id: 'password',
          type:'password'
        },
        value: '',
        label: 'Password',
        shouldValidate: true
    }
  let wrapper;
    beforeEach( ()=>{
      wrapper = shallow( <Input {...props} />)
    })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render input element', () =>{
    expect(wrapper.find('input')).toHaveLength(1)
  })

  it('should render select element', () => {
    const props = {
      elementType: 'select',
        elementConfig: {
          options: [
            {value: '', displayValue: 'Select...'},
            {value: 'test1', displayValue: 'Test1'},
            {value: 'test1', displayValue: 'Test2'},
          ],
          id: 'test'
        },
        value: 'test1',
        label: 'Test Select',
        validation: {},
        valid: true,
        changed: () => {}
    }
    wrapper = shallow(<Input {...props} />)
    expect(wrapper.find('option')).toHaveLength(3)
  })
  it('should render textarea', () => {
    const props = {
      elementType: 'textarea',
        elementConfig: {
          placeholder: 'Password',
          id: 'password',
          type:'password'
        },
        value: '',
        label: 'Password',
        shouldValidate: true,
        changed: () => {}
    }
    wrapper = shallow(<Input {...props} />);
    expect(wrapper.find('textarea')).toHaveLength(1)
  })
})