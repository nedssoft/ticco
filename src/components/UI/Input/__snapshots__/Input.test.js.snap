// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly 1`] = `
<div
  className="Input"
>
  <label
    className="Label"
    htmlFor="password"
  >
     
    Password
  </label>
  <input
    className="InputElement"
    id="password"
    placeholder="Password"
    type="password"
    value=""
  />
</div>
`;
